libnestutil/propagator_stability.cpp

```diff
@@ -28,43 +28,115 @@
+  : tau_syn_( 0.0 )
+  , tau_( 0.0 )
+  , c_m_( 0.0 )
+  , alpha_( 0.0 )
+  , beta_( 0.0 )
+  , gamma_( 0.0 )
+{
+}
+
+PropagatorExp::PropagatorExp( double tau_syn, double tau, double c_m )
+  : tau_syn_( tau_syn )
+  , tau_( tau )
+  , c_m_( c_m )
+  , alpha_( 1 / ( c_m * tau * tau ) * ( tau_syn - tau ) )
+  , beta_( tau_syn * tau / ( tau - tau_syn ) )
+  , gamma_( beta_ / c_m )
 {
-  const double P31_linear = 1 / ( 3. * C * tau * tau ) * h * h * h * ( tau_syn - tau ) * std::exp( -h / tau );
-  const double P31 = 1 / C
-    * ( std::exp( -h / tau_syn ) * numerics::expm1( -h / tau + h / tau_syn ) / ( tau / tau_syn - 1 ) * tau
-      - h * std::exp( -h / tau_syn ) )
-    / ( -1 - -tau / tau_syn ) * tau;
-  const double P31_singular = h * h / 2 / C * std::exp( -h / tau );
-  const double dev_P31 = std::abs( P31 - P31_singular );
-
-  if ( tau == tau_syn or ( std::abs( tau - tau_syn ) < 0.1 and dev_P31 > 2 * std::abs( P31_linear ) ) )
-  {
-    return P31_singular;
-  }
-  else
-  {
-    return P31;
-  }
 }
 
 double
-propagator_32( double tau_syn, double tau, double C, double h )
+PropagatorExp::evaluate( double h ) const
 {
-  const double P32_linear = 1 / ( 2. * C * tau * tau ) * h * h * ( tau_syn - tau ) * std::exp( -h / tau );
-  const double P32_singular = h / C * std::exp( -h / tau );
-  const double P32 =
-    -tau / ( C * ( 1 - tau / tau_syn ) ) * std::exp( -h / tau_syn ) * numerics::expm1( h * ( 1 / tau_syn - 1 / tau ) );
+  const double exp_h_tau_syn = std::exp( -h / tau_syn_ );
+  const double expm1_h_tau = numerics::expm1( -h / tau_ + h / tau_syn_ );
 
-  const double dev_P32 = std::abs( P32 - P32_singular );
+  double P32 = gamma_ * exp_h_tau_syn * expm1_h_tau;
 
-  if ( tau == tau_syn or ( std::abs( tau - tau_syn ) < 0.1 and dev_P32 > 2 * std::abs( P32_linear ) ) )
+  if ( tau_ == tau_syn_ or std::abs( tau_ - tau_syn_ ) < 0.1 )
   {
-    return P32_singular;
+    const double exp_h_tau = std::exp( -h / tau_ );
+
+    const double P32_singular = h / c_m_ * exp_h_tau;
+    if ( tau_ == tau_syn_ )
+    {
+      P32 = P32_singular;
+    }
+    else
+    {
+      const double P32_linear = alpha_ * h * h * exp_h_tau / 2.;
+      const double dev_P32 = std::abs( P32 - P32_singular );
+
+      if ( dev_P32 > 2 * std::abs( P32_linear ) )
+      {
+        P32 = P32_singular;
+      }
+    }
   }
-  else
+
+  return P32;
+}
+
+PropagatorAlpha::PropagatorAlpha()
+  : PropagatorExp( 0., 0., 0. )
+{
+}
+
+PropagatorAlpha::PropagatorAlpha( double tau_syn, double tau, double c )
+  : PropagatorExp( tau_syn, tau, c )
+{
+}
+
+std::tuple< double, double >
+PropagatorAlpha::evaluate( double h ) const
+{
+  const double exp_h_tau_syn = std::exp( -h / tau_syn_ );
+  const double expm1_h_tau = numerics::expm1( -h / tau_ + h / tau_syn_ );
+
+  double P31 = gamma_ * exp_h_tau_syn * ( beta_ * expm1_h_tau - h );
+  double P32 = gamma_ * exp_h_tau_syn * expm1_h_tau;
+
+  if ( tau_ == tau_syn_ or std::abs( tau_ - tau_syn_ ) < 0.1 )
```

Comment:
Looking at the code here again, I think it does not make much sense to have the first comparison, since it is included in the second part, and I assume that we do not save much by avoiding the subtraction. So
```suggestion
  if ( std::abs( tau_ - tau_syn_ ) < 0.1 )
```